src/server/PreviewServer.ts

```diff
@@ -109,12 +110,17 @@ export class PreviewServer {
       this.log('establishing server connection...')
 
       const connection = this.app.listen(port, host, () => {
-        const { address, port } = connection.address() as AddressInfo
-        const url = `http://${address}:${port}`
+        const { address, port, family } = connection.address() as AddressInfo
+        // IPv6 host requires surrounding square brackets when serialized to URL
+        // note: IPv6 host can take many forms, e.g. `::` and `::1` are both ok
+        const url = `http://${
+          family === 'IPv6' ? `[${address}]` : address
+        }:${port}`
         this.connectionInfo = {
           port,
           host: address,
           url,
+          family,
```

Comment:
Who/how is this `family` be consumed further down the chain?

Reply:
Ironically, the code that could benefit from it in https://github.com/mswjs/interceptors/pull/283 is the patch to `page-info` that won't be necessary once this is merged 😸 I think the patch for `@open-draft/test-server` could also use it, but I don't remember the exact code path. 

Basically, any logic that needs to differentiate between IPv6 and IPv4 based on the URL currently "guesses" with logic like `isIPv6 = address.includes(":")`. So returning the family here simply preserves information that we already have available from the `AddressInfo` object returned by `listen`. 

It's not strictly necessary but makes life easier downstream and shouldn't break anything. I guess it could break somebody's snapshot tests or something, but that would be surprising (especially since `this.connectionInfo` is never actually _returned_ - in the patch, I accessed it by exporting the mutable `server` in the browser integration tests setup function).

If you want to remove this, feel free to revert this commit (95f78f00df7b0546214a6dfe12647d7b3254e452) (or let me know your decision and I can force push the change).

Reply:
@milesrichardson, thanks for all your work on this. I'm also addressing related things in https://github.com/mswjs/interceptors/pull/354.

Reply:
Awesome! Sorry for all the back and forth (or no forth, as the case may be :eyes:). Let me know if I can help.

I haven't tried the latest version of msw in my Node 18 project yet (still using the hacky tarball I built), but I'll try switching it soon. (btw, that project is now open source, if you want to see how msw is used there: https://github.com/splitgraph/madatdata)

Reply:
Last I remember, everything was good to go. There were some failing tests, but the reason was because of a memory leak in the test runner, and the solution (which I didn't implement) would be to split it up into two calls to the test runner (I couldn't fix the memory leak after two days of trying).

I'm assuming you won't be using the code I wrote directly, so maybe that won't matter to you, but you can at least be assured that the latest commit in each respective project was known to be working at least once. Feel free to copy paste 😅

Reply:
I think this diff should contain the few necessary fixes to IPv6 serialization in msw-interceptors: https://github.com/milesforks/msw-interceptors/pull/6/files

And this is where the bulk of the work on `msw` was. This is the one where tests are only failing because of the memory leak, but otherwise everything else works. There is some IPv6 stuff in here too: https://github.com/milesforks/msw/pull/2/files